### DISCUSSION TITLE
Why a portable exe instead of windows pulsar zip?

### DISCUSSION THREAD

**Discussion by crimsonduelist** [pulsar-edit/.github#155] [General]
Why a self extracting exe instead of a zip?
From what I gather that s what the portable binary is -a portableapps-like exe.
I do see zip checked in https://github.com/orgs/pulsar-edit/discussions/77 and mac already has a zip version which can be run directly after setting up as portable https://pulsar-edit.dev/docs/launch-manual/sections/getting-started/#portable-mode

Is it just not there yet (and maybe not a priority at the moment)? I m guessing the dlls in the portable exe are responsible for setting up some things properly? i.e. u cant just go in and extract \app-64.7z\

**Comment by confused-Techie** [comment]
Really it's just never come up as something that's needed.

We use `electron-builder` to handle building all the binaries for Pulsar, so I'm not aware if it has the ability to export a proper zip file, but if it does it is something that can be added, we just haven't seen a need, since nobody has mentioned it up until now.

But I've now just checked and it is possible with `electron-builder` to set some targets that seem relevant like `7z`, `zip`, `tar.xz`, `tar.lz`, `tar.gz`, `tar.bz2`, and `dir`.

I'm assuming from the above `zip` would most likely be what you're requesting here?

**Comment by crimsonduelist** [comment]
zip or 7z is what I had in mind yes. I dont mind either.
-For what it s worth I use 7z where it s available but mostly zip is what I see available, Not sure if people have a preference between the 2 for whatever reason.

**Comment by Daeraxa** [comment]
Essentially it would be the same as the Linux tarball I suppose, just a "naked" binary and supporting files in a compressed folder. Personally I'd like to see the portable version developed somewhat - for the same reasons as supporting XDG base spec. It means we could natively support "portable mode" automatically - i.e. you launch the portable version and it will make a config dir in the same place you are launching it from to make it truly portable with no extra config.

> i.e. u cant just go in and extract \app-64.7z\

Can you not? I just tried it and it seems to be working ok? Unless there is a difference on first time use.

**Comment by crimsonduelist** [comment]
> Can you not? I just tried it and it seems to be working ok? Unless there is a difference on first time use.
>
It does seem to work indeed. -quick check with voidtools-everything nothing seems to ve been setup outside of the folder apart from temps/roaming I guess
I just extracted the \app-64.7z\ i.e. with 7z and setup as per [portable mode](https://pulsar-edit.dev/docs/launch-manual/sections/getting-started/#portable-mode) and [notes](https://pulsar-edit.dev/docs/launch-manual/sections/getting-started/#portable-notes)

> Essentially it would be the same as the Linux tarball I suppose, just a "naked" binary and supporting files in a compressed folder. Personally I'd like to see the portable version developed somewhat - for the same reasons as supporting XDG base spec. It means we could natively support "portable mode" automatically - i.e. you launch the portable version and it will make a config dir in the same place you are launching it from to make it truly portable with no extra config.
>
I see, but if nothing else is being set with the portable exe. u d be packaging an exe that just extracts and creates the .pulsar with some sub folders(which will already be taken care of on first run if .pulsar is there). Seems kinda wastefull

Still ll keep an eye out for a 7z/zip build. I do understand though other things taking precedence like updates/plugin support

**Comment by confused-Techie** [comment]
@crimsonduelist FYI This should now be fixed on our most recent `1.104.0` release if you wanna check it out

**Comment by crimsonduelist** [comment]
indeed!!
